File: core/src/main/java/forpleuvoir/dhwu1a/core/message/TempMessage.java
package forpleuvoir.dhwu1a.core.message;

import com.google.gson.JsonObject;
import forpleuvoir.dhwu1a.core.message.base.Message;
import forpleuvoir.dhwu1a.core.message.base.MessageType;
import forpleuvoir.dhwu1a.core.message.messagesender.TempMessageSender;
import forpleuvoir.dhwu1a.core.user.Group;
import forpleuvoir.dhwu1a.core.user.Member;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message
 * <p>#class_name TempMessage
 * <p>#create_time 2021/7/1 21:37
 */
public class TempMessage extends Message {
    /**
     * 发送消息的群员
     */
    public final Member member;
    /**
     * 相关的群
     */
    public final Group group;

    public TempMessage(JsonObject object) {
        super(MessageType.TempMessage, object);
        this.member = (Member) sender.getUser();
        this.group = ((TempMessageSender) sender).group.getUser();
    }

    @Override
    public void printMessageLog() {
        log.info("RM/Bot.{}:[{}({})] {}(Temp {}) >> {}", bot.id, group.data.name, group.id, member.data.memberName,
                 member.id, this.toPlainText()
        );
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/TempMessage.kt
package forpleuvoir.dhwu1a.core.message

import com.google.gson.JsonObject
import forpleuvoir.dhwu1a.core.message.base.Message
import forpleuvoir.dhwu1a.core.message.base.MessageType
import forpleuvoir.dhwu1a.core.message.messagesender.TempMessageSender
import forpleuvoir.dhwu1a.core.user.Group
import forpleuvoir.dhwu1a.core.user.Member

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message
 *
 * #class_name TempMessage
 *
 * #create_time 2021/7/1 21:37
 */
class TempMessage(`object`: JsonObject) : Message(MessageType.TempMessage, `object`) {
    /**
     * 发送消息的群员
     */
    @JvmField
    val member: Member? = sender.user as Member?

    /**
     * 相关的群
     */
    val group: Group = (sender as TempMessageSender).group!!.user
    override fun printMessageLog() {
        log.info(
            "RM/Bot.{}:[{}({})] {}(Temp {}) >> {}", bot.id, group.data.name, group.id, member!!.data.memberName,
            member.id, this.toPlainText()
        )
    }

}